### ISSUE TITLE
-

### ISSUE THREAD

**Issue by mglmc** [golangci/golangci-lint#6801]
### Summary

Running `golangci-lint run` repeatedly against the exact same, unmodified source tree produces a different set of `goconst` findings each time — same total finding *count*, different *strings and locations* flagged. This makes `goconst` unusable as a CI merge gate: the same commit can pass or fail depending on which run CI happens to pick.

### Environment

- `golangci-lint has version 2.13.2 built with go1.27.0 from 27774aaf on 2026-08-27T23:01:12Z` (official `golangci-lint-2.13.2-linux-amd64.tar.gz` release binary)
- Target module: `go 1.26.0` / `toolchain go1.26.6`, linux/amd64, 4 vCPUs
- Repo: a private, multi-package Go monorepo (~26 sibling packages under one `internal/domain/models/games/` tree), several packages containing data tables / word lists with many repeated short string literals (category labels, word-list entries, log event-type strings) — the kind of code `goconst` is meant to flag, just spread across many packages.
- `.golangci.yml` (relevant bit):
  ```yaml
  version: "2"
  linters:
    enable:
      - goconst
      - gocyclo
      - gosec
      - misspell
      - mnd
      - revive
    settings:
      goconst:
        min-occurrences: 3
  ```

### Repro

```bash
golangci-lint cache clean
golangci-lint run --timeout=10m > run1.txt
golangci-lint cache clean
golangci-lint run --timeout=10m > run2.txt
diff <(grep -oP "string \`[^\`]+\` has \d+ occurrences" run1.txt | sort -u) \
     <(grep -oP "string \`[^\`]+\` has \d+ occurrences" run2.txt | sort -u)
```

### Observed

Three consecutive runs against **identical, unmodified code**, cache cleaned before each:

| run | goconst count | example directories flagged |
|---|---|---|
| 1 | 50 | `crossclue` (39), `mindmeld` (8), `officecraze` (3), `cmd/simbot-server` (1, unrelated gosec) |
| 2 | 50 | `crossclue` (39), `guessdoodle` (5), `ws` (2), `officecraze` (2), `cavemanchat` (2) |
| 3 | 50 | `mindmeld` (43), `officecraze` (3), `ws` (2), `cavemanchat` (2) |

The total `goconst` count is stable at 50 every time, but the **union of distinct flagged strings across just those 3 runs is 77** — meaning individual runs disagree not only on *where* a repeated literal gets reported, but on *which literals* cross the `min-occurrences: 3` threshold at all.

Reproduced with:
- default concurrency and `-j 1` (rules out goroutine-scheduling races in the runner itself)
- a freshly cleaned `golangci-lint cache clean` before every single run (rules out stale/corrupt analysis cache)

So the non-determinism appears to live inside `goconst`'s own literal-occurrence tracking (possibly a `map` iteration order affecting which occurrences get counted/kept when there are many candidate literals across many packages), not in caching or scheduling.

### Expected

`goconst` (like the rest of `golangci-lint`) should produce the same findings on the same, unchanged input every time.

### Impact

We hit this while pinning `golangci-lint-action`/`golangci-lint` up from v2.5.0 (needed a version that declares `go 1.26.0` in its own `go.mod`, since v2.5.0 refuses to lint a `go1.26`-targeted module — "the Go language version (go1.25) used to build golangci-lint is lower than the targeted Go version (1.26.6)"). Bumping straight to v2.13.2 to fix that surfaced this separately. We're holding off on the version bump specifically because of this flakiness, not the version-mismatch issue.

Happy to share a minimal reproduction repo if useful — the above is drawn from a private codebase, so exact file contents aren't included here, but the shape (many sibling packages, each with 3+ repeated short string literals in data tables) should be easy to reconstruct.
